### DISCUSSION TITLE
sf update stable -a command shows later versions than latest stable release

### DISCUSSION THREAD

**Discussion by kez-sophos** [forcedotcom/cli#2732] [Q&A]
We are looking for a way to know if there is an update available for sf cli before triggering sf update. Running sf update stable -a command shows versions later than the latest stable release available when running sf update. Is this expected output for the command?

Sample output:
![image](https://github.com/forcedotcom/cli/assets/142229883/52dc958b-693e-4f2a-aec6-1870b88bb728)

Latest version as of posting is 2.28.6 (Feb 14, 2024) [stable]

**Comment by kez-sophos** [comment]
Update: 

![image](https://github.com/forcedotcom/cli/assets/142229883/026a7cda-0fa1-476a-bd46-558a92dd63ae)

Looks like I misunderstood how the sf update command works and there isn't actually a "sf update stable -a" command.

Is there a way to know if there is an update available within the sf cli and if not would something like that be useful?

**Comment by kez-sophos** [comment]
One can use the sf whatsnew command to check if there is an update available.

**Comment by mshanemc** [comment]
Yes.  We ship a [nightly release, and have a Release Candidate (RC)](https://developer.salesforce.com/docs/atlas.en-us.sfdx_setup.meta/sfdx_setup/sfdx_setup_install_cli_rc.htm) that's the next week's release.  So by the time something gets to `stable` there's been ~13 releases before that.

Besides the release notes and `sf whatsnew`,  you can see what the versions are by running
`npm view @salesforce/cli` (see `nightly` `latest-rc` and `latest` tags) or by looking on npm https://www.npmjs.com/package/@salesforce/cli?activeTab=versions
